.github/workflows/docs-release.yml

```diff
@@ -18,13 +18,24 @@ env:
 jobs:
   docs:
     runs-on: ubuntu-latest
-    container:
-      image: "ghcr.io/${{ github.repository_owner }}/dowhy-docs-generation:latest"
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gcc g++ pkg-config libgraphviz-dev graphviz-dev graphviz nodejs npm pandoc
+
+      - name: Install Poetry
+        run: python3 -m pip install poetry
+
       - name: Install Python Dependencies  
         run: poetry install -E plotting -E pydot -E pygraphviz -E econml --with docs
 
```

Comment:
@bloebp do we need to remove this git command about "--add safe.directory" here too? I saw that it was removed in docs-ci for the py313 branch.